src/components/CollectionsAndPalettes/common/RoundButton/RoundButton.tsx

```diff
@@ -1,45 +1,155 @@
+  activeMode?: string;
+  activeDisplay?: string;
   background?: string;
+  iconClass?: any;
+  hasAnimation?: boolean;
+  toggleMode?: any;
+  toggleDisplay?: any;
 }
 
-const activeButtons = {
+const activeButtonURLS = {
   palettes: PALETTES_URL,
   collections: COLLECTIONS_URL,
+  image: COLLECTIONS_IMAGE_URL,
+  info: COLLECTIONS_INFO_URL,
+  edit: COLLECTIONS_EDIT_URL,
 };
 
 export default class RoundButton extends React.Component<Props, any> {
-  static contextType = AppContext;
-  context!: AppContextState;
+  activeButtonMode: number;
+  animate = null;
+  animationDirection = null;
+
+  constructor(props) {
+    super(props);
+    switch (this.props.activeMode) {
+      case "image": {
+        this.activeButtonMode = 0;
+        break;
+      }
+      case "info": {
+        this.activeButtonMode = 1;
+        break;
+      }
+      case "edit": {
+        this.activeButtonMode = 2;
+        break;
+      }
+    }
+  }
+
+  switchDisplay(e, index, button) {
+    this.props.toggleDisplay(button);
+  }
+
+  toggleAnimation(e, index, button) {
+    this.animate = "true";
+    this.props.toggleMode(button);
+    if (this.activeButtonMode - index == 2) {
+      this.animationDirection = "animated-2steps-left";
+    } else if (this.activeButtonMode - index == -2) {
+      this.animationDirection = "animated-2steps-right";
+    } else if (this.activeButtonMode > index) {
+      this.animationDirection = "animated-left";
+    } else if (this.activeButtonMode < index) {
+      this.animationDirection = "animated-right";
+    } else {
+      this.activeButtonMode = null;
+    }
+    this.activeButtonMode = index;
+  }
+
+  renderDisplayButtons(index, button, buttonStyling) {
+    return (
+      <Link
+        to={COLLECTIONS_URL}
+        onClick={(e) => this.switchDisplay(e, index, button)}
+      >
+        <div
+          className={cn("button collection", buttonStyling[button])}
+          key={index}
+        >
+          {button}
+        </div>
+      </Link>
+    );
+  }
+
+  renderModeButtons(index, button, buttonStyling) {
+    return (
+      <Link
+        to={activeButtonURLS[button]}
+        onClick={(e) => this.toggleAnimation(e, index, button)}
+      >
+        <div
+          className={cn(
+            `${this.animationDirection}`,
+            "button collection",
+            buttonStyling[button]
+          )}
+          key={index}
+        >
+          {this.props.iconClass && (
+            <i className={this.props.iconClass[index]}></i>
+          )}
+          {button}
+        </div>
+      </Link>
+    );
+  }
+  renderExploreButtons(index, button, buttonStyling) {
+    return (
+      <div
+        className={cn("button collection", buttonStyling[button])}
+        key={index}
+      >
+        {button}
+      </div>
+    );
+  }
 
   render() {
-    const activeButton = window.location.href.includes(COLLECTIONS_URL)
+    const activeButtonMode = window.location.href.includes(COLLECTIONS_URL)
```

Comment:
Related to the previous comment about moving logic. Ideally, a "Buttons" component should never have to do things like this, the parent should just pass it the buttons that it should render, along with the currently active one (if any) and the "Buttons" component should just handle rendering them, and call a function on the parent when something should change.